File: src/dataset/javascript/prime-factors_exercism-ThomasZumsteg/openAI/ChatGPT4_prime-factors_v1.js
function primeFactors(remainder) {
	/* Finds all prime factors of a number */
	const factors = [];
	let factor = 2;

	while (remainder % factor === 0) {
		factors.push(factor);
		remainder /= factor;
	}

	factor = 3;
	const sqrtLimit = Math.sqrt(remainder);
	while (factor <= sqrtLimit) {
		while (remainder % factor === 0) {
			factors.push(factor);
			remainder /= factor;
		}
		factor += 2;
	}

	if (remainder > 1) factors.push(remainder);

	return factors;
}

module.exports = { for: primeFactors };

File: src/dataset/python/prime-factors_exercism-ThomasZumsteg/openAI/ChatGPT4_prime_factors_v1.py
"""Finds the prime factors of a number"""

def prime_factors(number):
    """Finds prime factors of a number"""
    factors = []
    while number % 2 == 0:
        factors.append(2)
        number //= 2
    for factor in range(3, int(number**0.5) + 1, 2):
        while number % factor == 0:
            factors.append(factor)
            number //= factor
    if number > 2:
        factors.append(number)
    return factors